slatedb/src/mem_table_flush.rs

```diff
@@ -155,6 +155,12 @@ impl MemtableFlusher {
                     modifier.state.manifest.core.recent_snapshot_min_seq =
                         min_active_snapshot_seq.unwrap_or(modifier.state.manifest.core.last_l0_seq);
 
+                    // update all the sequence numbers in the manifest
+                    let seq_tracker = self.db_inner.oracle.seq_tracker.read().map_err(|e| {
```

Comment:
We've been doing .expect() for lock poisoning. IMO, we should keep it that way for now (vs. introducing a new Internal error type).

Reply:
+1 for `.expect()` for poisoning.

btw, it seems we also had `parking_lot` included in the dependencies, i suppose we can also consider to replace this lock with a `parking_lot` for it removed poisoning in the lock apis.